src/shared/openssl-util.c

```diff
@@ -1128,6 +1128,97 @@ int string_hashsum(
         return 0;
 }
 #  endif
+
+static int load_key_from_provider(
+                const char *provider,
+                const char *private_key_uri,
+                EVP_PKEY **ret_private_key) {
+
+        assert(provider);
+        assert(private_key_uri);
+        assert(ret_private_key);
+
+#if OPENSSL_VERSION_MAJOR >= 3
+        if (!OSSL_PROVIDER_load(/* ctx= */ NULL, provider))
```

Comment:
doesn't matter much, but i think this should be try_load(), no? so that if this fails we don't disable openssl altogether